RestAPI.cs

```diff
@@ -43,12 +43,12 @@ public class RestAPI
             if (string.IsNullOrEmpty(url))
                 throw new Exception("Please use a valid WooCommerce Restful API url.");
 
-            url = url.Trim().ToLower().TrimEnd('/');

```

Comment:
the url i need is:  https://www.mywordpresssite/ABC/ internally the C# dll converts it to lower to this: https://www.mywordpresssite/abc/  which returns a 404 (Not found).  From looking at your code it looks like the ToLower() is so you can do some string compares, i created a separate variable so you can do the string compares without changing the actual url itself.